codes/extractive/netsum/netsum.py

```diff
@@ -0,0 +1,219 @@
+        
+        # for padding (manual)
+        if padding_needed:
+            for _ in range(len(text_sents),MAX_INPUT_SEQ_LENGTH):
+                label_list.append(0)
+                feature_list.append([0]*4)
+        
+        label.append(label_list)
+        feature.append(feature_list)
+            
+    # convert to numpy array
+    Y = np.array(label)
+    X = np.array(feature)
+    
+    if save_transformed_flag:
+        if transformed_path:
+            save_transformed_data(X, Y, transformed_path)
+            print("Transformed data has been saved under",transformed_path)
+        else:
+            # save transformed data in the working directory
+            save_transformed_data(X, Y, os.getcwd())
+            print("Transformed data has been saved under",os.getcwd())
+            
+    return X, Y
+
+def save_transformed_data(X, Y, saved_path):
+    X_file = h5py.File(os.path.join(saved_path,'features.h5'), 'w')
+    X_file.create_dataset('features', data=X)
+    X_file.close()
+    Y_file = h5py.File(os.path.join(saved_path,'labels.h5'), 'w')
+    Y_file.create_dataset('labels', data=Y)
+    Y_file.close()
+
+def load_transformed_data(X_path, Y_path):
+    X_file = h5py.File(X_path,'r')
+    X = X_file['features'][:]
+    X_file.close()
+    Y_file = h5py.File(Y_path,'r')
+    Y = Y_file['labels'][:]
+    Y_file.close()
+    return X, Y
+
+def rankNet(input_size, pretrained_weights=None):
+    inputs = Input(shape=input_size)
+    flat1 = Flatten()(inputs)
+    dense1 = Dense(128, activation='relu')(flat1)
+    dense2 = Dense(64, activation='relu')(dense1)
+    dense3 = Dense(32, activation='relu')(dense2)
+    dense4 = Dense(MAX_INPUT_SEQ_LENGTH, activation='softmax')(dense3)
+    
+    model = Model(input = inputs, output = dense4)
+    
+    # initiate RMSprop optimizer
+    opt = keras.optimizers.rmsprop(lr=0.0001, decay=1e-6)
+    
+    model.compile(loss='categorical_crossentropy',
+                  optimizer=opt,
+                  metrics=['accuracy'])
+    
+    print(model.summary())
+    
+    if(pretrained_weights):
+        model.load_weights(pretrained_weights)
+
+    return model
+    
+def training(x_train, y_train, x_test, y_test, filepath, pretrained_weights=None):
+    model = rankNet(x_train.shape[1:], pretrained_weights=pretrained_weights)
+    
+    # callback information
+    checkpoint = ModelCheckpoint(os.path.join(filepath,"ranknet_best_weights.h5"), monitor='val_loss', verbose=1, save_best_only=True, mode='min')
+    early_stopper = EarlyStopping(patience=5)
+    
+    model.fit(x_train, y_train,
+                  batch_size=BATCH_SIZE,
+                  epochs=EPOCHS,
+                  validation_data=(x_test, y_test),
+                  callbacks=[early_stopper, checkpoint],
+                  shuffle=True)
+    model.save_weights(os.path.join(filepath, 'rankNet.h5'))
+    
+def inference(x_test, y_test, pretrained_weights):
```

Comment:
do we use y test inside?

Reply:
True, no. I guess my original intention is to keep it here so that it could be used at the evaluation time (i.e. calculating Rogue 1 score)